src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -455,6 +460,10 @@ void CommissionerDiscoveryController::InternalHandleContentAppPasscodeResponse()
                     cd, Transport::PeerAddress::UDP(client->GetPeerAddress().GetIPAddress(), client->GetCdPort()));
                 return;
             }
+            // Store commissioner passcode as setup PIN (string/charspan).
+            // If this PIN is not empty it signals that user prompt was shown to enter PIN and AccountLogin::Login command has to be sent.
+            mCommissionerSetupPin = CharSpan::fromCharString(std::to_string(passcode));
```

Comment:
we already have local variable mPasscode so we should not need to capture this a second time (see calls to SetPasscode).

Reply:
But is this the same passcode? This case is when the mPasscode == 0 and this is storing the new passcode returned from GetCommissionerPasscode. Or are you saying it's OK to reuse mPasscode to store this new value?

Reply:
Removed this code in the refactoring. But please see updated question [here](https://github.com/project-chip/connectedhomeip/pull/34162/files#diff-828770158d3226d53556b9b513b02d618a1a493bedca2c8301a4a3ae3f90764cR639) after refactoring to clarify commissionee vs commissioner passcode.